src/bindgen/ir/ty.rs

```diff
@@ -255,9 +262,9 @@ impl Type {
                     Some(converted) => converted,
                     None => return Err("Cannot have an array of zero sized types.".to_owned()),
                 };
-
-                let path = GenericPath::load(&path.path)?;
-                let len = ArrayLength::Name(path.name);
+                // FIXMe: An array's length cannot really be generic, or can it?
```

Comment:
Kind of, I think you can reference associated constants of generic paths.

```rust
struct Foo<T> {
  data: T,
}
trait TraitConst {
  const Value: i32;
}
impl<T> TraitConst for Foo<T> {
  const Value: i32 = 1;
}
fn main() {
  let x = [i32; Foo::<i32>::Value as usize];
}
```

I don't think we support the actual semantic structure of rust paths, which are arbitrary sequences of idents, generic specifiers, and lifetime specifiers. So in the current state of things, it makes sense for this to just use `Path`.